edx/edx-enterprise
# [ENT-643] Direct enrollment to Audit Track

**Description:** This PR adds the ability to be directly enrolled into a course in audit mode through the course enrollment landing page, given some conditions:

1. The query parameter `audit` exists in the link.
1. The course is in the enterprise customer's catalog.

**JIRA:** [ENT-643](https://openedx.atlassian.net/browse/ENT-643)

**Merge deadline:** End of sprint

## Testing Instructions

1. Setup an Enterprise Customer with DSC enabled.
1. Go to the course enrollment landing page without any query parameters.
1. Enroll in the course in whatever mode you want, and confirm everything works as normal as before.
1. With an unenrolled user, go to the course enrollment landing page with the `audit` query parameter (it can take any value).
1. Confirm that you do not see the landing page, but see the DSC page immediately. Functionality from hereon should match the DSC workflows available in the `edx-platform`.
1. Setup the Enterprise Customer to disable DSC.
1. With an unerolled user, go to the landing page, but do not see the DSC page. Rather, you should be immediately directed to the LMS courseware page for that course.

#### To ensure no regressions:

1. Go to the program enrollment landing page and ensure our workflows are as normal. In particular, **nothing needs to be tested further** if you can confirm that the page renders correctly.

#### Segment

To test the fired track event to segment, we need to access the segment account linked to our business sandbox. This is located at https://segment.com/business-sandbox/sources/javascript/debugger -- ask the team if you need credentials.

1. Go to the course enrollment landing page with the `audit` query parameter attached. You can use something like https://business.sandbox.edx.org/enterprise/1b542b06-d353-45e6-9867-9579d9e12745/course/course-v1:edX+Test101+course/enroll/?audit=yes
1. See that once you get redirected to courseware (or DSC), an event is tracked at https://segment.com/business-sandbox/sources/javascript/debugger. Look for the event `edx.bi.user.enterprise.enrollment.course`. It should show something like this on the sidebar when you click it:

```python
analytics.track('394', 'edx.bi.user.enterprise.enrollment.course', {
    'category': 'enterprise',
    'enterprise_customer_name': 'Aviato',
    'enterprise_customer_uuid': '1b542b06-d353-45e6-9867-9579d9e12745',
    'label': 'course-v1:edX+Test101+course',
    'mode': 'audit'
})
```

---

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
